Library/Homebrew/tap.rb

```diff
@@ -871,27 +874,32 @@ def ==(other)
     other = Tap.fetch(other) if other.is_a?(String)
     self.class == other.class && name == other.name
   end
+  alias eql? ==
 
-  def self.each(&block)
-    return to_enum unless block
+  sig { returns(Integer) }
+  def hash
+    [self.class, name].hash
```

Comment:
Do we actually need the class here? It shouldn't be possible to get a `Tap` for e.g. `homebrew/core` anyways.

Reply:
It is considered good practice to include, in case this is ever used in mixed-type collections: https://ruby-doc.org/3.2.2/Object.html#method-i-hash